synapse/handlers/sync.py

```diff
@@ -765,6 +764,21 @@ def _generate_sync_entry_for_rooms(self, sync_result_builder, account_data_by_ro
             )
             sync_result_builder.now_token = now_token
 
+        # We check up front if anything has changed, if it hasn't then there is
+        # no point in going futher.
```

Comment:
It feels like that sort of comment would get out of date quickly, and wouldn't necessarily help that much.